src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,332 @@
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.Constants.CANIDs;
+import frc.robot.Constants.DIOConstants;
+import frc.robot.Robot;
+import utilities.DebugEntry;
+import utilities.TOFSensorSimple;
+import utilities.TOFSensorSimple.TOFType;
+
+public class IntakeSubsystem extends SubsystemBase {
+  /** Creates a new IntakeSubsystem. */
+  private TalonFX intakeMotor;
+
+  private TalonFXSimState simPivotMotor;
+
+  private TalonFX pivotMotor;
+  private TalonFX conveyorMotor;
+  private Angle pivotSetpoint = kPivotRetractAngle;
+
+  private DutyCycleEncoder throughBoreEncoder;
+
+  // Sensor closest to birdhouse
+  private TOFSensorSimple sensor;
+  private SimDeviceSim simSensor;
+  private SimBoolean simbeam;
+
+  private Mechanism2d mech = new Mechanism2d(2, 2);
+  private ComplexWidget widget;
+  private MechanismLigament2d pivotArmMech;
+
+  private enum IntakeState {
+    Idle,
+    atSensor,
+    coralLoaded
+  }
+
+  private IntakeState state = IntakeState.Idle;
+  private Timer timer = new Timer();
+
+  private SingleJointedArmSim pivotSim;
+
+  private DebugEntry<Double> pivotOutput;
+  private DebugEntry<String> currentCommand;
+
+  public IntakeSubsystem() {
+    intakeMotor = new TalonFX(CANIDs.kIntakeMotor);
+    pivotMotor = new TalonFX(CANIDs.kPivotMotor);
+    conveyorMotor = new TalonFX(CANIDs.kConveyorMotor);
+    sensor = new TOFSensorSimple(CANIDs.kConveyorSensor, Inches.of(1), TOFType.LASER_CAN);
+    throughBoreEncoder = new DutyCycleEncoder(DIOConstants.kthroughBoreEncoderID, 1, armZero);
+
+    /**
+     * Once the gains are configured, the Position closed loop control request can be sent to the
+     * TalonFX. The control request object has an optional feedforward term that can be used to add
+     * an arbitrary value to the output, which can be useful to account for the effects of gravity
+     * or friction.
+     *
+     * <p>// create a position closed-loop request, voltage output, slot 0 configs final
+     * PositionVoltage m_request = new PositionVoltage(0).withSlot(0);
+     *
+     * <p>// set position to 10 rotations m_talonFX.setControl(m_request.withPosition(10));
+     */
+    var slot0Configs = new Slot0Configs();
+    slot0Configs.kP = kPivotP;
+    slot0Configs.kI = kPivotI;
+    slot0Configs.kD = kPivotD;
+    slot0Configs.kG = kPivotG;
+    slot0Configs.kA = kPivotA;
+    slot0Configs.kV = kPivotV;
+    slot0Configs.GravityType = kPivotGravityType;
```

Comment:
It's not super important, but it could be work thinking about running SysID for the intake pivot. Idk how helpful it will be in this case, but something to note.
You could look at the docs to see how it works.

Reply:
Pushing adding SysId to a future PR